thirdparty/README.md

```diff
@@ -433,6 +433,18 @@ Files generated from upstream source:
 5. Copy `source/data/out/icudt76l.dat` to the `{GODOT_SOURCE}/thirdparty/icu4c/icudt76l.dat`
 
 
+## jolt_physics
+
+- Upstream: https://github.com/jrouwe/JoltPhysics
+- Version: 5.2.1 (e3d3cdf644389b621914bb6e73d52ee3137591a7, 2024)
+- License: MIT
+
+Files extracted from upstream source:
+
+- All files in `Jolt/` except `Jolt.cmake`
```

Comment:
We usually try to only use the files we use/compile, so since we're not using the ObjectStream files, I'd suggest removing them from the PR, and documenting here that the `ObjectStream` subfolder shouldn't be included.

And then you can remove the commented out files in the `SCsub` (maybe keeping the comment you have explaining why we don't include those files).

---

A subsidiary question is: aside from ObjectStream, do we compile/use all files from Jolt? If not, would it make sense to remove more of them, or would that be too much hassle?

The linker can be somewhat trusted to drop unused code so it shouldn't affect binary size, but compiling everything does affect compile time, and including all files (including their diffs on subsequent updates) impacts repo size.

Not a blocker for this PR in any case.

Reply:
> It's not the entire `ObjectStream` folder, as `ObjectStream/SerializableObject.cpp` is still needed, so I assume I would still need to document the individual files?

I guess then you can just document something like "Only keep `SerializableObject.{cpp,h}` from `ObjectStream`, instead of listing all other files to remove.

Reply:
I added a mention in `thirdparty/README.md` to look at the list of files under `ENABLE_OBJECT_STREAM`. I assume anyone looking to update Jolt should always consult that documentation anyway, and the list of files could theoretically be different inbetween updates, so perhaps that's sufficient and the comments in `modules/jolt_physics/SCsub` can be removed entirely?

Reply:
Yeah sounds good to me.